docs2/site/docs/migrations/migration7.md

```diff
@@ -136,6 +136,85 @@ the graph type to be specified in order to define a field.
 As with the expression syntax or the `AutoRegisteringObjectGraphType`,
 CLR type mappings can be tailored via the `schema.RegisterTypeMapping()` methods.
 
+### 10. Interface graph types can be automatically built from CLR types
+
+Similar to how input and output types can be inferred from their CLR counterparts,
+now interface graph types can also be inferred from CLR types. This is possible
+with the new class `AutoRegisteringInterfaceGraphType<TSourceType>` which functions
+identically to `AutoRegisteringObjectGraphType<TSourceType>` except creates an
+interface type rather than an object graph type. When using automatic CLR type
+mapping provided by `AddAutoClrMappings()` or `AddAutoSchema()`, any CLR interface
+type is automatically mapped to a interface graph type rather than an object graph
+type.
+
+Note that auto-mapped CLR types do not automatically register or link
+to any GraphQL interfaces; such mapping needs to be specified via the new
+`ImplementsAttribute`. Similarly, CLR types not referenced directly in the schema
+need to be added to the schema manually or else no graph type will be generated for them.
+
+Below is a typcial example of how the new functionality can be used:
+
+```csharp
+services.AddGraphQL(b => b
+    .AddAutoSchema<SampleQuery>()
+    .AddSystemTextJson());
+
+public class SampleQuery
+{
+    public static IAnimal Find(AnimalType type) => type switch
+    {
+        AnimalType.Cat => Cat(),
+        AnimalType.Dog => Dog(),
+        _ => throw new ArgumentOutOfRangeException(nameof(type)),
+    };
+
+    public static Cat Cat() => new Cat() { Name = "Fluffy", Lives = 9 };
+    public static Dog Dog() => new Dog() { Name = "Shadow", IsLarge = true };
+}
+
+public interface IObject
+{
+    [Id] int Id { get; }
+}
+
+public interface IAnimal : IObject
+{
+    AnimalType Type { get; }
+    string Name { get; }
+}
+
+public enum AnimalType { Cat, Dog }
+
+[Implements(typeof(IAnimal))]
+public class Cat : IAnimal
+{
+    [Id] public int Id => 10;
```

Comment:
Shouldn't the attribute be inherited to not duplicate it here?

Reply:
> It is important to ensure that any GraphQL attributes applied to members of the CLR types
are applied to both the interface and the classes alike, as the GraphQL.NET engine will build
distinct graph types for the interface and classes which implement those interfaces.

Well, I agree that GraphQL.NET engine will build distinct graph types but it _can_ use metainformation provided via interface/base class. By default all attributes have `Inherited = true`.

Reply:
I mean `GetPropertyInfo("Id").GetCustomAttributes` do return 'Id' attribute. Am I wrong?

Reply:
I tested it.  Your attribute inheritance works for derived classes, but not classes which implement interfaces.